cabal-install/Distribution/Client/Setup.hs

```diff
@@ -1424,6 +1430,13 @@ installOptions showOrParseArgs =
           installOnlyDeps (\v flags -> flags { installOnlyDeps = v })
           (yesNoOpt showOrParseArgs)
 
+      , option [] ["index-state"]
+          "Operate on package index state rolled back to given unix-timestamp"
```

Comment:
"Use the source package index as it existed at a previous time." and then extend this message with examples of the time formats accepted.